lib/rouge/lexers/cisco_ios.rb

```diff
@@ -0,0 +1,83 @@
+# -*- coding: utf-8 -*- #
+# frozen_string_literal: true
+
+# Based on/regexes mostly from Brandon Bennett's pygments-routerlexers:
+# https://github.com/nemith/pygments-routerlexers
+
+module Rouge
+  module Lexers
+    class CiscoIos < RegexLexer
+      title 'Cisco IOS'
+      desc 'Cisco IOS configuration lexer'
+      tag 'cisco_ios'
+      filenames '*.cfg'
+      mimetypes 'text/x-cisco-conf'
+
+      state :root do
+        rule %r/^!.*/, Comment::Single
+
+        rule %r/^(version\s+)(.*)$/ do
+          groups Keyword, Num::Float
+        end
+
+        rule %r/(desc*r*i*p*t*i*o*n*)(.*?)$/ do
```

Comment:
Can you really use `desccccrrrriipppptttttinnnn` as a keyword?

Reply:
Maybe you mean something like `%r/desc(r(i(p(t(i(on?)?)?)?)?)?/`?

Reply:
@chapmajs Can we match the exact text, ie. `description`? 🤔 If the user input the wrong text,  this rule will be simply skipped through.

Reply:
It would be preferable to keep the partial match that's currently there -- many Cisco IOS commands and keywords can be shortened, and this helps highlight e.g. console session captures with the same consistency as highlighting an actual `show running-config` output. Unfortunately, it's not sufficient to match just a shortened version, as IOS will happily take any level of completion from the minimum required for unique identification to the full command/keyword.

Reply:
For example, the above `show running-config` can be (and often is):

* `sh run`
* `show run`
* `sho running`
* `show runn`

Reply:
I think I accidentally clicked the "Resolve conversation" button last night, rather than hitting the reply box as intended.